dandiapi/api/views/dandiset.py

```diff
@@ -538,3 +598,36 @@ def clear_uploads(self, request, dandiset__pk):
 
         dandiset.uploads.all().delete()
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+    @swagger_auto_schema(
+        methods=['POST'],
+        manual_parameters=[DANDISET_PK_PARAM],
+        request_body=no_body,
+        responses={
+            200: 'Dandiset starred successfully',
+            401: 'Authentication required',
+        },
+        operation_summary='Star a dandiset.',
+        operation_description='Star a dandiset. User must be authenticated.',
+    )
+    @action(methods=['POST'], detail=True)
+    def star(self, request, dandiset__pk) -> Response:
+        dandiset = self.get_object()
+        star_count = star_dandiset(user=request.user, dandiset=dandiset)
+        return Response({'count': star_count}, status=status.HTTP_200_OK)
+
+    @swagger_auto_schema(
+        manual_parameters=[DANDISET_PK_PARAM],
+        request_body=no_body,
+        responses={
+            200: 'Dandiset unstarred successfully',
+            401: 'Authentication required',
+        },
+        operation_summary='Unstar a dandiset.',
+        operation_description='Unstar a dandiset. User must be authenticated.',
+    )
+    @star.mapping.delete
```

Comment:
What does this decorator do?

Reply:
This decorator tells DRF to mount the function under the same endpoint as the `star`, using the `DELETE` method.  However, you raise a good point, these functions differ only slightly, and should be unified.